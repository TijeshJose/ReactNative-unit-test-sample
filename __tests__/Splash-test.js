import 'react-native';
import React from 'react';
import Splash from '../src/splash';

import renderer from 'react-test-renderer';
import AsyncStorage from '@react-native-community/async-storage';

//check splash page renders correctly
it('Splash page renders correctly', () => {
    const tree = renderer.create(<Splash />).toJSON();
    expect(tree).toMatchSnapshot();
});


//check with mock async storage value
it('checks if Async Storage is used', async () => {
    await AsyncStorage.getItem('email');

    expect(await AsyncStorage.getItem).toBeCalledWith('email');
})