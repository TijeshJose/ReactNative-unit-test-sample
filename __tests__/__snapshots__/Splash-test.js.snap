// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Splash page renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#999999"
    hidesWhenStopped={true}
    size="small"
  />
</View>
`;
